effectsize_example.Rmd

```diff
@@ -85,7 +100,7 @@ library(ggstance)   # for geom_pointrangeh()
 
 ```{r boilerplate, include = FALSE}
 set.seed(12)
-format_num <- function(...) format(..., digits = 2)
+format_num <- function(x, k = 2) format(round(x, k), nsmall = k)
```

Comment:
I spent some more time trying to figure out R formatting functions, because I didn't really like the one I or Chat had suggested. The current one is this:

```{r}
format_num <- function(x, k = 2) format(round(x, k), nsmall = k)
```

This formatting function will:

1. always have two trailing digits, even if the input number has no decimal places (e.g. format_num(1) -> "1.00") or is very large (e.g. format_num(12345) -> "12345.00")
2. round anything less than 0.005 to 0 (e.g. format_num(0.005) -> "0.00")

A more direct version (without the need for `round()`) might be:

```{r}
format_num <- function(x, digits = 2) formatC(x, digits, format = "f")
```

(this gives the same result as using `round` above without the additional `round`ing step by switching `format` to `formatC` and employing `format = "f"`, which changed the interpretation of the `digits` argument).

Although, I'm not sure we want criteria 1 above (always have trailing 0s), so I would suggest at least:

```{r}
format_num <- function(x, digits = 2) formatC(x, digits, format = "f", drop0trailing = TRUE)
```

Which gives format_num(1) -> "1" and format_num(12345) -> "12345", but format_num(1.2345) -> "1.23"

On the other hand, I was taught with [significant digits](https://en.wikipedia.org/wiki/Significant_figures) that the relevant thing is not an absolute number of digits, but the number of digits from the leftmost position, which is what the `digits` argument to `formatC` does, perhaps like this:

```{r}
format_num <- function(x, digits = 3) formatC(x, digits = 3, format = "g", width = -1)
```

Which gives:

- format_num(1) -> "1"
- format_num(12345) -> "123e+04"
- format_num(1.2345) -> "1.23"
- format_num(0.0012345) -> "0.00123"

Though I'm not sure we want the scientific notation in there for 12345, so maybe:

```{r}
format_num <- function(x, digits = 3) formatC(x, digits = 3, format = "fg", width = -1)
```

- format_num(1) -> "1"
- format_num(12345) -> "12345"
- format_num(1.2345) -> "1.23"
- format_num(0.0012345) -> "0.00123"